tests/functional/adapter/hooks/test_run_hooks.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+from dbt.tests.adapter.hooks import test_run_hooks as core_base
+
+
+class TestPrePostRunHooksBigQuery(core_base.TestPrePostRunHooks):
+    def check_hooks(self, state, project, host):
+        self.get_ctx_vars(state, project)
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            # The create and drop table statements here validate that these hooks run
+            # in the same order that they are defined. Drop before create is an error.
+            # Also check that the table does not exist below.
+            "on-run-start": [
```

Comment:
Lots of jinja here. Perhaps set these as separate variables like we do with models and schemas?